### DISCUSSION TITLE
Building from the command line / MSBuild does not work.

### DISCUSSION THREAD

**Discussion by todor-dk** [cefsharp/CefSharp#3994] [General]
## Version / System

- NuGet CefSharp.Wpf.NETCore 97.1.60
- x64
- Net 6.0
- Win 10
- WPF

## Test Project
1. In VS 2022, create a new **WPF Application** project on .Net 6.
2. Add references to CefSharp in the .csproj file:
```
<Project Sdk="Microsoft.NET.Sdk">
	<PropertyGroup>
		<OutputType>WinExe</OutputType>
		<TargetFramework>net6.0-windows</TargetFramework>
		<UseWPF>true</UseWPF>
		<Platforms>AnyCPU;x64</Platforms>
	</PropertyGroup>
	<ItemGroup>
		<PackageReference Include="CefSharp.Wpf.NETCore" Version="97.1.60" />
	</ItemGroup>
</Project>
```    
3. Add a web browser to MainWindow.xaml.
`<wpf:ChromiumWebBrowser Address="http://www.google.com" />`
4. In the solution configuration manager, create a new solution platform **x64** and delete the existing **Any CPU** platform.
4. Build and run. Everything OK!

## Reproduce the Problem
1. Delete the **obj** and **bin** folders (to ensure everything is clean).
2. In the test project's solution directory, build using MSBuild. I used the **Developer Command Prompt for VS 2022**, MSBuild version 17.0.0.52104.
`msbuild.exe /nologo /m /nr:false /p:Configuration="Debug" /p:Platform=x64 /t:restore;build`
3. Run the app. It should not start, as some files are missing.

## Expected
That the building from MSBuild will produce the same results as from within Visual Studio.

## Additional Information
After long debugging, I've concluded the following:
- If compiled from within VS, the **cefsharp.common.netcore\97.1.60\buildTransitive\CefSharp.Common.NETCore.targets** dynamically adds the following package references:
```
<PackageReference Include="chromiumembeddedframework.runtime.win-x64" Version="97.1.6" />
<PackageReference Include="chromiumembeddedframework.runtime.win-x86" Version="97.1.6" />
<PackageReference Include="chromiumembeddedframework.runtime.win-arm64" Version="97.1.6" />
```
- If building with MSBuild, the above package references are missing. This is visible if one compares the contents of the **obj** folder, especially the **project.assets.json**. If built from VS, the file is different than if built from the command line.
- It appears that the restore phase of MSBuild does not process the **CefSharp.Common.NETCore.targets** correctly and imports the package references.
- A work around to make MSBuild work is to modify the test project to manually include the native runtime packages. This is not feasible in a complex solution with many transitive project dependencies.
```
<Project Sdk="Microsoft.NET.Sdk">
	<ItemGroup>
		<PackageReference Include="CefSharp.Wpf.NETCore" Version="97.1.60" />
		<PackageReference Include="chromiumembeddedframework.runtime.win-x64" Version="97.1.6" />
		<PackageReference Include="chromiumembeddedframework.runtime.win-x86" Version="97.1.6" />
		<PackageReference Include="chromiumembeddedframework.runtime.win-arm64" Version="97.1.6" />
	</ItemGroup>
</Project>
```

**Comment by amaitland** [comment]
Converting to discussion as this is to be best of my knowledge is outside my direct control. 

Ill provide additional detail when time permits.

**Reply by amaitland** [reply]
I won't have time today to go into this in any detail.

The short answer is you need to specify a [RuntimeIdentifier](https://docs.microsoft.com/en-us/dotnet/core/rid-catalog)

```
msbuild /restore /t:Publish /p:RuntimeIdentifier=win-x64 /p:configuration=release
```

So add `/p:RuntimeIdentifier=win-x64` to your `msbuild args`.

I don't know if `msbuild` sets `SelfContained` by default, so you might need to set that to false also.
